File: CSES/1070.py
n = int(input())
ans = []

if n == 1:
    print("1")
if n == 2 or n == 3:
    print("NO SOLUTION")
if n >= 4:
    for i in range(2, n+1, 2):
        ans.append(str(i))
    for i in range(1, n+1, 2):
        ans.append(str(i))

    print(" ".join(ans))

File: CSES/1070.cpp
#include<iostream>
using namespace std;

int main(){
    int n;
    cin >> n;

    if (n == 1){
        cout << "1";
    }

    if (n == 2 || n == 3){
        cout << "NO SOLUTION";
    }

    if (n >= 4){
        for (int i=2; i<=n; i+=2){
            cout << i << " ";
        }
        for (int i=1; i<=n; i+=2){
            cout << i;
            if (n - 1  < n){
                cout << " ";
            }
        }
    }
}